services/src/main/java/io/druid/cli/EmbeddedBroker.java

```diff
@@ -0,0 +1,188 @@
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.inject.Binder;
+import com.google.inject.Inject;
+import com.google.inject.Injector;
+import com.google.inject.Key;
+import com.google.inject.Module;
+import com.metamx.common.guava.Sequence;
+import com.metamx.common.lifecycle.Lifecycle;
+import com.metamx.common.logger.Logger;
+import io.druid.client.BrokerSegmentWatcherConfig;
+import io.druid.client.BrokerServerView;
+import io.druid.client.CachingClusteredClient;
+import io.druid.client.TimelineServerView;
+import io.druid.client.cache.CacheConfig;
+import io.druid.client.cache.CacheMonitor;
+import io.druid.client.selector.CustomTierSelectorStrategyConfig;
+import io.druid.client.selector.ServerSelectorStrategy;
+import io.druid.client.selector.TierSelectorStrategy;
+import io.druid.guice.CacheModule;
+import io.druid.guice.GuiceInjectors;
+import io.druid.guice.Jerseys;
+import io.druid.guice.JsonConfigProvider;
+import io.druid.guice.LazySingleton;
+import io.druid.guice.LifecycleModule;
+import io.druid.guice.annotations.Self;
+import io.druid.query.MapQueryToolChestWarehouse;
+import io.druid.query.Query;
+import io.druid.query.QueryContextKeys;
+import io.druid.query.QuerySegmentWalker;
+import io.druid.query.QueryToolChestWarehouse;
+import io.druid.query.RetryQueryRunnerConfig;
+import io.druid.server.ClientQuerySegmentWalker;
+import io.druid.server.DruidNode;
+import io.druid.server.QueryManager;
+import io.druid.server.QueryResource;
+import io.druid.server.initialization.ServerConfig;
+import io.druid.server.initialization.jetty.JettyServerInitializer;
+import io.druid.server.metrics.MetricsModule;
+import io.druid.server.router.TieredBrokerConfig;
+import org.eclipse.jetty.server.Server;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ */
+public class EmbeddedBroker extends ServerRunnable
+{
+  private static final Logger log = new Logger(EmbeddedBroker.class);
+
+  public EmbeddedBroker()
+  {
+    super(log);
+  }
+
+  @Override
+  protected List<? extends Module> getModules()
+  {
+    return ImmutableList.of(
+        new Module()
+        {
+          @Override
+          public void configure(Binder binder)
+          {
+            JsonConfigProvider.bindInstance(
+                binder, Key.get(DruidNode.class, Self.class),
+                new DruidNode(TieredBrokerConfig.DEFAULT_BROKER_SERVICE_NAME, null, null)
+            );
+
+            binder.bind(QueryToolChestWarehouse.class).to(MapQueryToolChestWarehouse.class);
+
+            binder.bind(CachingClusteredClient.class).in(LazySingleton.class);
+            binder.bind(BrokerServerView.class).in(LazySingleton.class);
+            binder.bind(TimelineServerView.class).to(BrokerServerView.class).in(LazySingleton.class);
+
+            JsonConfigProvider.bind(binder, "druid.broker.cache", CacheConfig.class);
```

Comment:
How would you feel about making these properties `druid.embedded.broker.` or similar?